presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/TestPushAggregationThroughOuterJoin.java

```diff
@@ -67,14 +53,9 @@ public void testPushesAggregationThroughLeftJoin()
                                         Optional.empty(),
                                         Optional.empty(),
                                         Optional.empty()
-                                ),
-                                ImmutableMap.of(p.symbol("AVG", DOUBLE),
-                                        new AggregationNode.Aggregation(averageFunctionCall, avgSignature, Optional.empty())),
-                                ImmutableList.of(ImmutableList.of(p.symbol("COL1", BIGINT))),
-                                SINGLE,
-                                Optional.empty(),
-                                Optional.empty())
-                )
+                                ))
+                        .addAggregation(p.symbol("AVG", DOUBLE), PlanBuilder.expression("avg(COL2)"), ImmutableList.of(DOUBLE))
+                        .groupingSets(ImmutableList.of(ImmutableList.of(p.symbol("COL1", BIGINT))))))
```

Comment:
or maybe we could call a `groupingSet` method per each grouping set and then builder would collect them internally
`
.groupingSet(p.symbol("C1"))
.groupingSet(p.symbol("C2"), p.symbol("C1")
`

How does it sound to you?